### PR TITLE
Eature/eui 2187 integrate xui node and common lib

### PR DESCRIPTION
**Pull request by paddybasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/application.ts [hunk 1]
```diff
@@ -1,30 +1,26 @@
 import * as bodyParser from 'body-parser'
 import * as cookieParser from 'cookie-parser'
 import * as express from 'express'
-import * as session from 'express-session'
 import * as helmet from 'helmet'
 // import {router as termsAndCRoutes} from './termsAndConditions/routes'
```
Comment (rayliangatos) [inline api/application.ts:5]: though this is not part of the changes, can these unuse imports be tidied up

File: api/auth/index.ts [hunk 2]
```diff
@@ -1,69 +1,130 @@
...
+                s2sEndpointUrl: `${getConfigValue(SERVICE_S2S_PATH)}/lease`,
+                s2sSecret: s2sSecret.trim(),
+            },
+        },
+        session: showFeature(FEATURE_REDIS_ENABLED) ? redisStoreOptions : fileStoreOptions,
+    }
+
+    const type = showFeature(FEATURE_OIDC_ENABLED) ? 'oidc' : 'oauth2'
+    nodeLibOptions.auth[type] = options
+    logger._logger.info('Setting XuiNodeLib options')
```
Comment (rayliangatos) [inline api/auth/index.ts:128]: can this logger._logger be simplified?

File: api/lib/proxy.ts [hunk 3]
```diff
@@ -12,38 +12,43 @@ import {
...
 
 import { http } from './http'
-//import * as log4jui from './log4jui'
 import { EnhancedRequest } from './models'
+import {exists} from './util'
 
-//const logger = log4jui.getLogger('proxy')
-
-export function setHeaders(req: EnhancedRequest) {
+export function setHeaders(req: express.Request) {
```
Comment (rayliangatos) [inline api/lib/proxy.ts:18]: can this be simplified to Request

File: api/lib/proxy.ts [hunk 4]
```diff
@@ -12,38 +12,43 @@ import {
...
+        if (exists(req, 'headers.ServiceAuthorization')) {
+            headers.ServiceAuthorization = req.headers.ServiceAuthorization
+        }
     }
 
     return headers
 }
 
-export async function get(req: EnhancedRequest, res: express.Response, next: express.NextFunction) {
+export async function get(req: express.Request, res: express.Response, next: express.NextFunction) {
```
Comment (rayliangatos) [inline api/lib/proxy.ts:51]: can this be simplified to Request, Response, NextFunction?

File: src/app/services/auth/auth.service.ts [hunk 5]
```diff
@@ -1,106 +1,38 @@
...
+  public logOut(): Observable<any> {
+    return this.httpService.get('/auth/logout?noredirect=true');
   }
 
-  public getWindowLocation() {
-    return window.location;
+  public logOutAndRedirect() {
+    this.logOut().subscribe( () => {
+      this.setWindowLocationHref('/idle-sign-out')
+    })
```
Comment (rayliangatos) [inline src/app/services/auth/auth.service.ts:35]: missing ; which has tslint issue

File: api/user/index.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { getUserSessionTimeout } from './userTimeout'
 
 export async function getUserDetails(req, res: Response, next: NextFunction) {
 
-  const { roles } = req.session.user
+  const { roles } = req.session.passport.user.userinfo
```
Comment (rayliangatos) [inline api/user/index.ts:8]: this looks awkard any reason to wrap with passport

File: api/application.ts (hunk 1 above)
Comment (paddybasi) [reply api/application.ts:5]: we intend to put T&C's back in once it's been deployed to production

File: api/auth/index.ts (hunk 2 above)
Comment (paddybasi) [reply api/auth/index.ts:128]: This is how the 'old' logger was coded, exposing the underlying logger as _logger (I prefer to use this as it's not destructive, the old one would convert objects to [Object Object])

File: api/lib/proxy.ts (hunk 3 above)
Comment (paddybasi) [reply api/lib/proxy.ts:18]: Ideally we need a clean up task of the node layer, including removing of ```EnhancedRequest``` however this PR isn't the place for it :)

File: api/lib/proxy.ts (hunk 4 above)
Comment (paddybasi) [reply api/lib/proxy.ts:51]: Yes it can, again a clean up task (tech debt for later)

File: api/user/index.ts (hunk 6 above)
Comment (paddybasi) [reply api/user/index.ts:8]: the library we've pushed uses passport under the hood; if we then stored user back onto ```req.session``` there would be a lot of data that would have to be transferred back/forth to redis.

File: src/app/services/auth/auth.service.ts (hunk 5 above)
Comment (paddybasi) [reply src/app/services/auth/auth.service.ts]: fixed, though it would only have given a warning (which we have plenty of anyway :joy: )

**Pull request merged**